packages/xdl/src/Exp.js

```diff
@@ -284,16 +284,20 @@ export async function getPublishInfoAsync(root: string): Promise<PublishInfo> {
   const remotePackageName = name;
   const remoteUsername = username;
   const remoteFullPackageName = `@${remoteUsername}/${remotePackageName}`;
-  const bundleIdentifierIOS = exp.ios ? exp.ios.bundleIdentifier : null;
+
+  const platformSpecific =
+    platform === 'android'
+      ? { androidPackage: _.get(exp, 'android.package', null) }
+      : { bundleIdentifierIOS: _.get(exp, 'ios.bundleIdentifier', null) };
```

Comment:
To keep things simple, I suggest we just always return both of these fields (or `null` if they're not defined in the manifest) and remove the optional `platform` argument.